File: perl/button_test.pl
#!/usr/bin/perl

use strict;
use warnings;

use Time::HiRes qw(sleep);
use HiPi::Device::GPIO;
use HiPi::Constant qw( :raspberry );

my $dev = HiPi::Device::GPIO->new();
my $pin = $dev->export_pin( RPI_PAD1_PIN_12 );
$pin->mode( RPI_PINMODE_INPT );

while(1) {
	unless($pin->value) {
		print "Pressed\n";
		sleep .2;
	}
}

File: python/button_test.py
#!/usr/bin/python
import time
import RPi.GPIO as GPIO
GPIO.setmode(GPIO.BOARD)
GPIO.setup(12, GPIO.IN)

while True:
	mybutton = GPIO.input(12)
	if mybutton == False:
		print "Pressed"
		time.sleep(.2)
